conkit/io/consurf.py

```diff
@@ -0,0 +1,31 @@
+from conkit.io._parser import PredictionFileParser
+from conkit.core.predictionfile import PredictionFile
+from conkit.core.residueprediction import ResiduePrediction
+
+
+class ConsurfParser(PredictionFileParser):
+    """Class for parsing consurf files."""
+
+    def read(self, f_handle, f_id="consurf"):
+        """Return consurf conservation scores instance."""
+
+        hierarchy = PredictionFile(f_id, predtype = 'conservationpred')
+
+        for line in f_handle:
+            line = line.split()
+            if len(line) >= 1 and line[0].isnumeric():
+                residue = ResiduePrediction(line[0])
+                if line[3].replace('*', '').isnumeric():
+                    residue.prediction = int(line[3].replace('*', ''))
+                    hierarchy.add(residue)
+                else:
+                    print('Consurf cannot be parsed. Conflict in line %s' % ''.join(line))
+                    exit()
```

Comment:
what is this? you are exiting with an interpreter function.

1. you should use ``sys.exit`` if you want to do this.
2. exit should contain a non-zero error code, i.e ``sys.exit(1)``, to indicate an issue
3. **raise an ``Exception``, it is not in the parser's power to decide when to exit**